src/array_util.rb

```diff
@@ -0,0 +1,43 @@
+class ArrayUtil
```

Comment:
Strange name for a class, why did you choose to have the methods in a class?

Reply:
I used to have classes for better code management (in big projects).

Reply:
But this is not a big project, is always better to adapt to the problem.

Checkout how Matz (the creator of ruby) uses ruby to solve a simple problem https://www.youtube.com/watch?v=zEf6iUIkjf4